File: Vue-frontend/src/types/note.ts
/** Capture mode for sidebar notes. Legacy notes omit this field. */
export type NoteType = 'excerpt' | 'quick' | 'question'

/** AI organization lifecycle. Legacy notes omit this field. */
export type OrganizeStatus = 'pending' | 'organizing' | 'organized' | 'error'

/** Automatic source kinds limited to learning surfaces. */
export type NoteSourceType = 'plan' | 'resource' | 'knowledge_tree' | 'tutor'

export interface Note {
  id: number
  userId: number
  noteName: string
  content: string
  tags?: string[] | string
  isPinned?: number
  noteType?: NoteType
  organizeStatus?: OrganizeStatus
  sourceType?: NoteSourceType
  sourceId?: number
  sourceTitle?: string
  sourceRoute?: string
  excerpt?: string
  createdAt: string
  updatedAt: string
}

export interface NoteResourceRel {
  id: number
  noteId: number
  resourceId: number
  selectedText?: string
  positionInfo?: string
  planId?: number
  moduleName?: string
  resourceTitle?: string
}

export interface NoteCreateRequest {
  noteName: string
  content: string
  tags?: string[]
  isPinned?: number
  noteType?: NoteType
  organizeStatus?: OrganizeStatus
  sourceType?: NoteSourceType
  sourceId?: number
  sourceTitle?: string
  sourceRoute?: string
  excerpt?: string
}

export interface NoteLinkRequest {
  resourceId: number
  selectedText: string
  positionInfo: string
  planId?: number
  moduleName?: string
  resourceTitle?: string
}

File: Android-app/app/src/main/java/com/aura/mas/data/model/Note.kt
package com.aura.mas.data.model

import com.google.gson.annotations.SerializedName

data class Note(
    val id: Long = 0,
    @SerializedName("userId") val userId: Long = 0,
    @SerializedName("noteName") val noteName: String = "",
    val content: String = "",
    @SerializedName("isDeleted") val isDeleted: Int = 0,
    val createdAt: String? = null,
    val updatedAt: String? = null
)

data class NoteCreateRequest(
    @SerializedName("noteName") val noteName: String,
    val content: String = ""
)

data class NoteResourceRel(
    val id: Long = 0,
    @SerializedName("noteId") val noteId: Long = 0,
    @SerializedName("resourceId") val resourceId: Long = 0,
    @SerializedName("selectedText") val selectedText: String = "",
    @SerializedName("planId") val planId: Long? = null,
    @SerializedName("moduleName") val moduleName: String = "",
    @SerializedName("resourceTitle") val resourceTitle: String = ""
)
